src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,39 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder builder = new StringBuilder("Report for period ");
+        builder.append(dateFrom).append(" - ").append(dateTo);
+        LocalDate localDateFrom = LocalDate.parse(dateFrom, formatter);
+        LocalDate localDateTo = LocalDate.parse(dateTo, formatter);
+        for (String name : names) {
+            builder.append(System.lineSeparator()).append(name).append(" - ");
+            int salary = 0;
+            for (String dataElement : data) {
+                String[] dataPerDataElementArray = dataElement.split(" ");
+                LocalDate localData = LocalDate.parse(dataPerDataElementArray[0], formatter);
```

Comment:
```suggestion
                LocalDate localData = LocalDate.parse(dataPerDataElementArray[DATE_INDEX], formatter);
```
![image](https://github.com/mate-academy/jv-salary-info/assets/20326858/3455bb5f-b3cd-489d-8d39-686a1ba428fd)